kiagnose/internal/checkup/namespace/namespace.go

```diff
@@ -44,30 +44,36 @@ func Create(client kubernetes.Interface, namespace *corev1.Namespace) (*corev1.N
 
 // DeleteAndWait delete and waits for the given namespace to dispose.
 func DeleteAndWait(client kubernetes.Interface, nsName string, timeout time.Duration) error {
-	if err := client.CoreV1().Namespaces().Delete(context.Background(), nsName, metav1.DeleteOptions{}); err != nil {
-		return err
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	nsLabel := fmt.Sprintf("%s=%s", corev1.LabelMetadataName, nsName)
+	nsWatcher, err := client.CoreV1().Namespaces().Watch(ctx, metav1.ListOptions{LabelSelector: nsLabel})
+	if err != nil {
+		return fmt.Errorf("failed to create namespace %q watcher: %v", nsName, err)
 	}
```

Comment:
Defer has to be done just after resource instantiation
```suggestion
	}
	defer nsWatcher.Close();
```

Reply:
You mean `Stop()`?